dts/bindings/pwm/nxp,flexio-pwm.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP Flexio PWM controller.
+  Each flexio timer can be used for generating one pwm pulse.
+  The two PWM modes supported by flexio are chosen based on the selected polarity -
+  Dual 8-bit counters PWM mode and Dual 8-bit counters PWM Low mode.
+
+compatible: "nxp,flexio-pwm"
+
+include: [pwm-controller.yaml, pinctrl-device.yaml, base.yaml]
+
+properties:
+  pinctrl-0:
+    required: true
+
+  pinctrl-names:
+    required: true
+
+  "#pwm-cells":
+    const: 3
+
+pwm-cells:
+  - channel
+  - period
+  - flags
+
+child-binding:
+  description: |
+    Flexio PWM channel configuration.
+
+  properties:
+    pin-id:
+      type: int
+      required: true
+      description: |
+        pin-id should be populated with number 'n' specified in the
+        FXIO_D<n> data pin of your board's schematic.
+
+    prescaler:
+      type: int
+      description: |
+        The clock divider for internal counter prescaler.
+        - 0: Divide by 1
+        - 4: Divide by 16
+        - 5: Divide by 256
+      enum: [0, 4, 5]
```

Comment:
Flexio IP's Timer configuration register has some bits which configures the source of the timer decrement and the source of the shift clock. The bits required for PWM configuration can be configured as 0, 4 or 5. 
My primary concern is to configure the IP register  with that. That's why I have considered these values.

Reply:
> I'd prefer for the enum to be [1, 16, 256] and resolve that to the [0, 4, 5] at compile-time

I agree with this. By doing that we can get rid of `pwm_nxp_flexio_get_prescaler_div()` and resolve the value with `_CONCAT(FLEXIO_PWM_CLK_DIV_, DT_INST_PROP(n, prescaler))`

Reply:
Thanks @manuargue 
Made this change, but since there is no direct logic to convert - 0,4,5 to 1,16,256 so I had to add another config member prescaler_div.
A better suggestion is most welcome..